internal/webhook/certs/cert_test.go

```diff
@@ -0,0 +1,358 @@
+package certs
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testSecretName    = "test-secret"
+	testNamespaceName = "test-namespace"
+	testServiceName   = "test-service"
+)
+
+func Test_serviceAltNames(t *testing.T) {
+	type args struct {
+		serviceName string
+		namespace   string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "service AltNames are generated correctly",
+			args: args{serviceName: "test-service", namespace: "test-namespace"},
+			// not using consts here to make it as readable as possible.
+			want: []string{
+				"test-service.test-namespace.svc",
+				"test-service",
+				"test-service.test-namespace",
+				"test-service.test-namespace.svc.cluster.local",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := serviceAltNames(tt.args.serviceName, tt.args.namespace)
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
what would be the pros of this change? I'm not sure, because this file with tests was copied from the `function-controller`, but it looks like generated a scaffold by the IDE and it works fine.

Reply:
The test would be much easier to read, the flow is straight without any ifs.
The diff is included in assertion message. Less opportunity to make a typo.